src/main/java/org/apache/mesos/curator/CuratorStateStore.java

```diff
@@ -183,21 +186,25 @@ public void storeStatus(Protos.TaskStatus status) throws StateStoreException {
         // Validate that a TaskInfo with the exact same UUID is currently present. We intentionally
         // ignore TaskStatuses whose TaskID doesn't (exactly) match the current TaskInfo: We will
         // occasionally get these for stale tasks that have since been changed (with new UUIDs).
-        Protos.TaskInfo taskInfo;
+        Optional<Protos.TaskInfo> optionalTaskInfo;
         try {
-            taskInfo = fetchTask(taskName);
+            optionalTaskInfo = fetchTask(taskName);
         } catch (Exception e) {
             throw new StateStoreException(String.format(
-                    "Unable to retrieve matching TaskInfo for the provided TaskStatus name %s. " +
-                            "Call storeTasks() before calling storeStatus()", taskName), e);
+                    "Unable to retrieve matching TaskInfo for the provided TaskStatus name %s.", taskName), e);
+        }
+
+        if (!optionalTaskInfo.isPresent()) {
+            throw new StateStoreException("The following TaskStatus is not present in the StateStore: " + taskName);
```

Comment:
TaskInfo, not TaskStatus?

In the error, could add a blurb explaining that 'TaskInfo must be present before writing TaskStatus'